packages/sdk/src/conversation/sso/ssoExecutionDialog.ts

```diff
@@ -0,0 +1,307 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  ComponentDialog,
+  WaterfallDialog,
+  Dialog,
+  DialogSet,
+  DialogTurnStatus,
+  DialogContext,
+} from "botbuilder-dialogs";
+import {
+  Activity,
+  ActivityTypes,
+  Channels,
+  StatePropertyAccessor,
+  Storage,
+  tokenExchangeOperationName,
+  TurnContext,
+} from "botbuilder";
+import { CommandMessage, SsoExecutionDialogHandler, TriggerPatterns } from "../interface";
+import { TeamsBotSsoPrompt, TeamsBotSsoPromptSettings } from "../../bot/teamsBotSsoPrompt";
+import { TeamsBotSsoPromptTokenResponse } from "../../bot/teamsBotSsoPromptTokenResponse";
+import { TeamsFx } from "../../core/teamsfx";
+import { v4 as uuidv4 } from "uuid";
+import { formatString } from "../../util/utils";
+import { ErrorCode, ErrorMessage, ErrorWithCode } from "../../core/errors";
+import { internalLogger } from "../../util/logger";
+
+let DIALOG_NAME = "SsoExecutionDialog";
+let TEAMS_SSO_PROMPT_ID = "TeamsFxSsoPrompt";
+let COMMAND_ROUTE_DIALOG = "CommandRouteDialog";
+
+/**
+ * Sso execution dialog, use to handle sso command
+ */
+export class SsoExecutionDialog extends ComponentDialog {
+  private dedupStorage: Storage;
+  private dedupStorageKeys: string[] = [];
+  private commandMapping: Map<string, string | RegExp | (string | RegExp)[]> = new Map<
+    string,
```

Comment:
minor: may be add some comments to explain what does the key and value mean?

Reply:
sure